diff --git a/ios/Podfile b/ios/Podfile
index 6b1937a..9e119e6 100644
--- a/ios/Podfile
+++ b/ios/Podfile
@@ -47,6 +47,14 @@ target 'Jellyfin' do
         end
       end
     end
+
+    # This is necessary for Xcode 15 compatibility
+    # https://github.com/facebook/react-native/issues/37748#issuecomment-1580589448
+    installer.pods_project.targets.each do |target|
+      target.build_configurations.each do |config|
+        config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= ['$(inherited)', '_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION']
+      end
+    end
   end
 
   post_integrate do |installer|
diff --git a/ios/Podfile.lock b/ios/Podfile.lock
index bea2dad..c4b90aa 100644
--- a/ios/Podfile.lock
+++ b/ios/Podfile.lock
@@ -632,6 +632,6 @@ SPEC CHECKSUMS:
   RNScreens: 4a1af06327774490d97342c00aee0c2bafb497b7
   Yoga: 7a4d48cfb35dfa542151e615fa73c1a0d88caf21
 
-PODFILE CHECKSUM: 886347fc7958cd8169a4e06f4a5e125dadf419a3
+PODFILE CHECKSUM: 6ff4c838a1c209b25f53e9709460e3847fe29a5e
 
 COCOAPODS: 1.15.2
